src/iexamine.cpp

```diff
@@ -3431,6 +3432,144 @@ void iexamine::climb_down( player &p, const tripoint &examp )
 }
 
 /**
+ * Use base command to edit options and utalize base functions.
+ * 
+ * TODO: Finish writting. Change to uimenu.
+ */
+void iexamine::base_control( player &p, const tripoint &examp) {
+    #define base (p->home) 
```

Comment:
This looks like a bad approach. The player should not "own" a base, rather the base should "be owned" by the player.
You can save owner's ID in base structure - player has a `getID` (I might have misspelled it) function.

Reply:
And please don't use macros like this. You can create a "local alias" via references:
```C++
auto &base = p->home;
```

Reply:
Since it's already getting passed a reference to the player it seemed easier to just add a reference to the base in player.